wp_api/src/request/endpoint.rs

```diff
@@ -78,6 +78,10 @@ impl TryFrom<&str> for ApiBaseUrl {
 }
 
 impl ApiBaseUrl {
+    pub fn with_api_url(url: Url) -> Self {
```

Comment:
We probably should drop the `impl From<Url> for ApiBaseUrl` implementation because having both of these might create confusion.

Looking back, I think the way `impl From<Url> for ApiBaseUrl` implemented wasn't a good idea in the first place. The `ApiBaseUrl` only has a `Url` field, so the expectation for `impl From<Url> for ApiBaseUrl` would be to set that field directly, instead of modifying it the way it's done at the moment.

Let me know if you want to address that in this PR or if you'd rather I open a PR against `trunk`.

Reply:
It's best to refactor it in a separate PR, which I can look into later.